src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -2453,7 +2453,7 @@ void java_lang_Throwable::print_stack_trace(Handle throwable, outputStream* st)
       BacktraceElement bte = iter.next(THREAD);
       print_stack_element_to_stream(st, bte._mirror, bte._method_id, bte._version, bte._bci, bte._name);
     }
-    {
+    if (THREAD->can_call_java()) {
```

Comment:
This allows `java_lang_Throwable::print_stack_trace` to be used in contexts where Java calls cannot be made.